File: LeetCode Solutions/Problems/2155. All Divisions With the Highest Score of a Binary Array.py
class Solution:
    def maxScoreIndices(self, nums: List[int]) -> List[int]:
        currentScore = sum(nums)
        currentMax = currentScore
        result = [0]
        for i in range(1, len(nums) + 1):
            currentScore += 1 - 2 * nums[i - 1]
            if currentScore > currentMax:
                currentMax = currentScore
                result = [i]
            elif currentScore == currentMax:
                result.append(i)
        return result

File: LeetCode Solutions/Problems/2155. All Divisions With the Highest Score of a Binary Array.java
class Solution {
    public List<Integer> maxScoreIndices(int[] nums) {
        int currentScore = Arrays.stream(nums).sum();
        int currentMax = currentScore;
        List<Integer> result = new ArrayList<>();
        result.add(0);
        for (int i=1; i<=nums.length; ++i){
            currentScore += 1 - 2 * nums[i - 1];
            if (currentScore > currentMax) {
                currentMax = currentScore;
                result.clear();
                result.add(i);
            } 
            else if (currentScore == currentMax){
                result.add(i);
            }
        }
        return result;
    }
}